File: app/src/main/java/es/osw/cafelatte/presentation/di/base/module/ActivityModule.java
package es.osw.cafelatte.presentation.di.base.module;

import dagger.Module;
import dagger.Provides;
import es.osw.cafelatte.presentation.ui.BaseActivity;

@Module public class ActivityModule {
  private final BaseActivity mActivity;

  public ActivityModule(BaseActivity activity) {
    this.mActivity = activity;
  }

  @Provides BaseActivity provideBaseActivity() {
    return this.mActivity;
  }
}

File: app/src/main/java/es/osw/cafelatte/presentation/di/base/module/ActivityModule.kt
package es.osw.cafelatte.presentation.di.base.module

import dagger.Module
import dagger.Provides
import es.osw.cafelatte.presentation.di.base.PerActivity
import es.osw.cafelatte.presentation.ui.BaseActivity

@PerActivity @Module class ActivityModule(private val activity: BaseActivity) {

    @Provides internal fun provideBaseActivity(): BaseActivity {
        return activity
    }
}
